### PR TITLE
Setup codefoafrica cms

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

- This PR sets up payload CMS for codeforafrica.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/payload.config.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { buildConfig } from "payload/config";
+import Pages from "./src/payload/collections/Pages";
+import { CollectionConfig } from "payload/types";
+import dotenv from "dotenv";
+
+dotenv.config();
+dotenv.config({ path: "./.env.local" });
+
+const appURL = process?.env?.CFA_PAYLOAD_PUBLIC_APP_URL;
```
Comment (kilemensi) [inline apps/codeforafrica/payload.config.ts:9]:
```suggestion
const appURL = process?.env?.PAYLOAD_PUBLIC_APP_URL;
```

or `PAYLOAD_PUBLIC_CFA_APP_URL` if we must. `PAYLOAD_PUBLIC_` prefix is required.

File: apps/codeforafrica/server.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import payload from "payload";
+import { loadEnvConfig } from "@next/env";
+
+const projectDir = process.cwd();
+loadEnvConfig(projectDir);
+
+const app = express();
+
+const hostname = process.env.NEXT_HOSTNAME || "localhost";
+const PORT = parseInt(process.env.PAYLOAD_PORT || "3010", 10);
```
Comment (kilemensi) [inline apps/codeforafrica/server.ts:15]:
```suggestion
const PORT = parseInt(process.env.PAYLOAD_PORT || "3000", 10);
```

Why `3010`?

File: apps/codeforafrica/server.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+const dev = process.env.NODE_ENV !== "production";
+
+const start = async () => {
+  try {
+    await payload.init({
+      secret: process.env.CFA_PAYLOAD_SECRET,
+      mongoURL: process.env.CFA_MONGODB_URI,
```
Comment (kilemensi) [inline apps/codeforafrica/server.ts:22]:
```suggestion
      secret: process.env.PAYLOAD_SECRET,
      mongoURL: process.env.MONGODB_URL,
```

File: apps/codeforafrica/payload.config.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        ...config.resolve,
+        fallback: {
+          ...config?.resolve?.fallback,
+          fs: false,
+          os: false,
+          "process/browser": false,
+        },
+      },
+    }),
+  },
```
Comment (kilemensi) [inline apps/codeforafrica/payload.config.ts:27]: 1. Include must have plugins: nested docs, seo, storage,
2. Include must have Media collection

File: apps/codeforafrica/src/payload/collections/Pages.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { pages } from "./slugNames";
+
+const Pages = {
+  slug: pages,
+  access: {
+    read: () => true,
+  },
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/collections/Pages.js:7]: Enable drafts/versions from get go.

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: apps/codeforafrica/src/payload/utils/formatDraftUrl.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+function formatDraftUrl(collection, doc, { locale }) {
+  const pagePath = `/${collection}/${doc.slug}`;
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/utils/formatDraftUrl.js:2]: Didn't we have a function to process this? For pages with parent, e.g. resources/datasets, the above code wouldn't work.

File: apps/codeforafrica/env.template [hunk 7]
```diff
@@ -22,3 +22,13 @@ GITHUB_BACKEND_REPO=
...
+
+# PAYLOAD
+PAYLOAD_PUBLIC_CFA_APP_URL=http://localhost:3010
+PAYLOAD_PUBLIC_PORT=3010
+PAYLOAD_SECRET=randomsecretkeythatyougenerateherethisisjustanexample
+MONGODB_URL=mongodb://root:rootpassword@localhost:27017/codeforafrica?authSource=admin
+#S3
+S3_ACCESS_KEY_ID=
+S3_SECRET_ACCESS_KEY=
+S3_ACCESS_KEY_ID=
```
[the remark this answers is not in the document]
Comment (kilemensi) [reply apps/codeforafrica/env.template:34]: Need a new line here.

**Comment by kilemensi** [comment]
Your review @kelvinkipruto

**Comment by kelvinkipruto** [comment]
@koechkevin Looks like you have to add `ts-node` dev dependency. I don't have it installed so starting the app fails.

File: apps/codeforafrica/src/payload/collections/Media.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  upload: {
+    staticURL: "/media",
+    staticDir: "media",
+  },
+  fields: [
+    {
+      name: "alt",
+      label: "Alt Text",
+      type: "text",
+      localized: true,
```
Comment (kelvinkipruto) [inline apps/codeforafrica/src/payload/collections/Media.js:18]: Will codeforAfrica site support multiple languages? If not, no need for localization

File: apps/codeforafrica/src/payload/collections/Pages.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
+import fullTitle from "../fields/fullTitle";
+import slug from "../fields/slug";
+import formatDraftUrl from "../utils/formatDraftUrl";
+
+import { pages } from "./slugNames";
+
+const Pages = {
+  slug: pages,
```
Comment (kelvinkipruto) [inline apps/codeforafrica/src/payload/collections/Pages.js:8]: Not sure this is necessary. We can just set it to "pages". No need to declare slug in a different file

File: apps/codeforafrica/package.json [hunk 10]
```diff
@@ -22,7 +22,10 @@
   },
   "scripts": {
     "cms": "netlify-cms-proxy-server",
-    "dev": "NODE_OPTIONS='--inspect' next dev --port 3002",
+    "build-server": "tsc --project tsconfig.server.json",
+    "build-next": "NEXT_BUILD=true pnpm pnpm build-server && NEXT_BUILD=true PAYLOAD_PUBLIC_APP_URL=http://localhost:3000 PAYLOAD_CONFIG_PATH=dist/payload.config.js node dist/server.js",
+    "build-payload": "payload build",
+    "dev": "NODE_OPTIONS='--inspect' ts-node --project tsconfig.server.json server.ts",
```
Comment (kelvinkipruto) [inline apps/codeforafrica/package.json:28]: add `ts-node` as a dev dependency

File: apps/codeforafrica/src/payload/utils/formatDraftUrl.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+
+function formatDraftUrl(collection, doc) {
+  const pagePath = formatPagePath(collection, doc);
+  if (pagePath) {
+    const slug = pagePath;
+    const url = new URL(
+      `/api/v1/draft?slug=${slug}`,
+      process.env.PAYLOAD_PUBLIC_CFA_APP_URL
+    );
+    console.error("......>>", url);
```
Comment (kelvinkipruto) [inline apps/codeforafrica/src/payload/utils/formatDraftUrl.js:11]: Remove `console` statements

**Review by kelvinkipruto — APPROVED** [review]

**Review by kelvinkipruto — APPROVED** [review]

**Review by kelvinkipruto — COMMENTED** [review]
@koechkevin When you give a page a parent, the page title does not include the parent

**Comment by kelvinkipruto** [comment]
> When you give a page a parent, the page title does not include the parent

@koechkevin It looks ok, I got confused a little bit between `title` and `full title`. Perhaps you should remove title and let full title remain like in charterAfrica

**Review by kelvinkipruto — APPROVED** [review]

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**